File: harrison_kreps/hk_price_singlebeliefs.py
"""

Authors: Chase Coleman, Tom Sargent

"""
import numpy as np
import scipy.linalg as la

def price_single_beliefs(transition, dividend_payoff, β=.75):
    """
    Function to Solve Single Beliefs
    """
    # First compute inverse piece
    imbq_inv = la.inv(np.eye(transition.shape[0]) - β*transition)

    # Next compute prices
    prices = β * np.dot(np.dot(imbq_inv, transition), dividend_payoff)

    return prices

File: harrison_kreps/hk_price_singlebeliefs.jl
#=

Authors: Shunsuke Hori

=#
"""
Function to Solve Single Beliefs
"""
function price_singlebeliefs(transition::Matrix, dividend_payoff::Vector;
                             beta::AbstractFloat=.75)
    # First compute inverse piece
    imbq_inv = inv(eye(size(transition,1)) - beta*transition)

    # Next compute prices
    prices = beta * ((imbq_inv*transition)* dividend_payoff)

    return prices
end
